### PR TITLE
Upgrade Docker Compose to v2.16.0

### PR DESCRIPTION
**Pull request by milas**
https://github.com/docker/compose/releases/tag/v2.16.0

### CODE REVIEW

File: runner/actions-runner-dind-rootless.ubuntu-22.04.dockerfile
Comment (mumoshu) [inline runner/actions-runner-dind-rootless.ubuntu-22.04.dockerfile:8]: Note: I do believe this is the right version used in GitHub-hosted Ubuntu 22.04 runners: https://github.com/actions/runner-images/blob/bcb2886e5b0fb66c491f66640461128755c06885/images/linux/Ubuntu2204-Readme.md?plain=1#L74

File: runner/actions-runner-dind-rootless.ubuntu-20.04.dockerfile
Comment (mumoshu) [inline runner/actions-runner-dind-rootless.ubuntu-20.04.dockerfile:8]: Note: This does correspond to what is used in GitHub-hosted Ubuntu 20.04 runners: https://github.com/actions/runner-images/blame/bcb2886e5b0fb66c491f66640461128755c06885/images/linux/Ubuntu2004-Readme.md?plain=1#L80

**Review by mumoshu — APPROVED** [review]
LGTM. Thank you so much for your contribution @milas!
Let me merge this once we get an additional approval from the Actions team.

**Review by Link- — APPROVED** [review]
LGTM 👍 thank you both!

**Pull request merged**
